bot/tests/plugins/test_cash_pool.py

```diff
@@ -0,0 +1,215 @@
+        self.db.reset_mock()
+        self.db_history.reset_mock()
+
+        # for fx in self.pool.response_fns:
+        #     fx.reset_mock()
+
+
+    @pytest.mark.parametrize('message,expected,history', [
+        # currencies
+        ('kevin paid $10 for dan', [
+            mock.call('kevin', ['dan'], 1000, 'cad'),
+        ], []),
+        ('kevin paid $10 cad for dan', [
+            mock.call('kevin', ['dan'], 1000, 'cad'),
+        ], []),
+        ('kevin paid $10 usd for dan', [
+            mock.call('kevin', ['dan'], 1000, 'usd'),
+        ], []),
+        ('kevin paid $10 aus for dan', [], []),
+
+        # reasons
+        ('for testing purposes kevin paid $1 for dan', [
+            mock.call('kevin', ['dan'], 100, 'cad'),
+        ], [
+            mock.call(mock.ANY, mock.ANY, mock.ANY, mock.ANY,
+                      'for testing purposes', 'my_name'),
+        ]),
+
+        # parse even split
+        ('kevin paid $10 for dan and lara', [
+            mock.call('kevin', ['dan', 'lara'], 1000, 'cad'),
+        ], []),
+        ('kevin paid $10 for dan, lara and steven', [
+            mock.call('kevin', ['dan', 'lara', 'steven'], 1000, 'cad'),
+        ], []),
+        ('kevin paid $10 for dan, lara, and steven', [
+            mock.call('kevin', ['dan', 'lara', 'steven'], 1000, 'cad'),
+        ], []),
+        ('kevin paid $10 for dan and lara and steven', [
+            mock.call('kevin', ['dan', 'lara', 'steven'], 1000, 'cad'),
+        ], []),
+
+        # parse multi-payer
+        ('kevin paid $10 and dan paid $15 for kevin', [
+            mock.call('kevin', ['kevin'], 1000, 'cad'),
+            mock.call('dan', ['kevin'], 1500, 'cad'),
+        ], []),
+        ('kevin paid $10 and dan paid $15 for kevin and dan', [
+            mock.call('kevin', ['kevin', 'dan'], 1000, 'cad'),
+            mock.call('dan', ['kevin', 'dan'], 1500, 'cad'),
+        ], []),
+        ('kevin paid $10 and dan paid $15 for kevin, dan, and lara', [
+            mock.call('kevin', ['kevin', 'dan', 'lara'], 1000, 'cad'),
+            mock.call('dan', ['kevin', 'dan', 'lara'], 1500, 'cad'),
+        ], []),
+        ('kevin paid $10, dan paid $15, and lara paid $20 '
+         'for kevin, lara, and dan', [
+             mock.call('kevin', ['kevin', 'lara', 'dan'], 1000, 'cad'),
+             mock.call('dan', ['kevin', 'lara', 'dan'], 1500, 'cad'),
+             mock.call('lara', ['kevin', 'lara', 'dan'], 2000, 'cad'),
+         ], []),
+
+        # parse uneven-split
+        ('kevin paid $10 for lara and kevin paid $15 for dan', [
+            mock.call('kevin', ['lara'], 1000, 'cad'),
+            mock.call('kevin', ['dan'], 1500, 'cad'),
+        ], []),
+        ('kevin paid $10 for lara and kevin paid $15 for dan and lara', [
+            mock.call('kevin', ['lara'], 1000, 'cad'),
+            mock.call('kevin', ['dan', 'lara'], 1500, 'cad'),
+        ], []),
+        ('kevin paid $15 for dan and lara and kevin paid $10 for lara', [
```

Comment:
I guess you still run into the issue "kevin paid $45 for Dan, Lara, Kevin paid $50 for Dan. We'd need people to use and at the end.